src/LogikUI/File/FileManager.cs

```diff
@@ -0,0 +1,337 @@
+                        else throw new InvalidProjectDataException($"Start ({ start.X }, { start.Y }) and end ({ end.X }, { end.Y }) of wire #{ Wires.Count + 1 } has to be on the same axis.");
+
+                        // FIXME: Diagonal wire support
+                    }
+                }
+
+                #endregion
+
+                #region Parse Components
+
+                foreach (var _component in doc.SelectNodes("/circuit/components"))
+                {
+                    if (_component != null)
+                    {
+                        XmlNode component = ((XmlNode)_component).FirstChild;
+
+                        ComponentType type = types[component.SelectSingleNode("type").InnerText];
+                        Vector2i location = getPos(component.SelectSingleNode("location"));
+                        Circuit.Orientation orientation = orientations[component.SelectSingleNode("orientation").InnerText];
+
+                        Components.Add(InstanceData.Create(type, location, orientation));
+
+                        // FIXME: Update list of gates.
+                    }
+                }
+
+                #endregion
+
+                #region Parse Labels
+
+                foreach (var _label in doc.SelectNodes("/circuit/labels"))
+                {
+                    if (_label != null)
+                    {
+                        XmlNode label = ((XmlNode)_label).FirstChild;
+
+                        int size = int.Parse(label.Attributes["size"].InnerText);
+                        Vector2i location = getPos(label.SelectSingleNode("location"));
+                        string text = label.SelectSingleNode("text").InnerText;
+
+                        Labels.Add(new TextLabel(location, text, size));
+                    }
+                }
+
+                #endregion
+
+                // FIXME: Save wires, components, and labels in higher level objects
+
+                IsNew = false;
+                FileManager.filename = filename;
+            }
+            catch (Exception e)
+            {
+                if (e is XmlException || e is XPathException || e is InvalidProjectDataException || e is XmlSchemaException || e is KeyNotFoundException)
+                {
+                    MessageDialog md = new MessageDialog(null, DialogFlags.DestroyWithParent, MessageType.Info, ButtonsType.Ok, "Logik is unable to read file. Please choose a valid project file.");
+                    md.Run();
+                    md.Dispose();
+
+                    throw new ArgumentException("Not a project file.", e);
+                }
+                else
+                {
+                    MessageDialog md = new MessageDialog(null, DialogFlags.DestroyWithParent, MessageType.Error, ButtonsType.Ok, "Logik is unable to access choosen file.");
+                    md.Run();
+                    md.Dispose();
+
+                    throw new ArgumentException("Unable to access file.", e);
+                }
```

Comment:
It's not guaranteed that not being able to access the file is the reason we get here...
I think we should give a more custom message for each different exception and include some data about exactly what happened.